config/configmodels/configmodels.go

```diff
@@ -44,15 +44,50 @@ type Config struct {
 type NamedEntity interface {
```

Comment:
The name of the interface no longer matches the functions it defines. This is not just a named entity anymore since it has Type and Enabled functions. We will need to rename the interface. However, do we actually need to touch this in this PR?